[Cohii2] iP

Cohii2 opened the pull request:
Pull Request

Cohii2 committed:
Rename Chatbot

To Noriaki

Cohii2 committed:
Greet and Exit

Implemented greet method

Cohii2 committed:
Modified greet function

Tidied up main

Cohii2 committed:
Created Task class

Cohii2 committed:
Echo method

Completed Echo method

Cohii2 committed:
Add, List

Overrode toString() operator in Task class for ease of printing description & completed Add functionality - new startList method

Cohii2 committed:
Resolved Warnings

Cohii2 committed:
Wrote comments

Cohii2 committed:
Updated List

anneleong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,164 @@
+    /**
+     * Prints goodbye message.
+     */
+    public static void goodbye(){
```
Names representing methods must be verbs and written in camelCase.
i think it would be better to name it as prinntGoodbyeMessage

anneleong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,164 @@
+    /**
+     * Prints list of tasks.
+     *
+     * @param taskList List of tasks to be printed.
```
the @param can be omitted since the variable taskList is quite self explanatory

anneleong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,164 @@
+        while(true){
+            String input = readInput();
+
+            if (input.equalsIgnoreCase("bye")) {
```
conditional case follows the coding standard yayy :)

anneleong commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,37 @@
+        this.isDone = done;
+    }
+
+//    @Override
```
would be better to remove comments that are not used in the actual code when u push your code to github

anneleong reviewed (commented):
keep up the good work :))

Geinzit reviewed (commented):
Overall very nice, each section is nicely commented and relatively easy to read! Good job!

Geinzit commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,164 @@
+     *
+     * @param message Message to be printed.
+     */
+    public static void printMessage(String message){
```
I like how you encapsulated repeating actions into single methods.

Geinzit commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,164 @@
+                " OK, I've marked this task as not done yet:\n" + "   [ ] "
+                + taskList.get(index).getDescription();
+
+        try {
```
For this catch exception it may be better to include the message part above inside the "try" section as well, since it could also lead to invalid index access and such.

Geinzit commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,37 @@
+        return description;
+    }
+
+    public boolean isDone() {
```
I thnk it's better to name a get function as what it is, for example, checkIsDone or getIsDone.

Geinzit reviewed (commented):
Overall very nice, each section is nicely commented and relatively easy to read! Good job!

Cohii2 committed:
Update input read

Update user input reading method to hopefully throw less exceptions and errors.

Cohii2 committed:
Fix task bugs

Add task now adds entire input, and mark and unmark task methods amend the correct tasks.

Cohii2 committed:
Implement Task types

Implemented ToDo, Event, Deadline.

Cohii2 committed:
Automated Text UI Testing

Use the input/output redirection technique to semi-automate the testing of Duke. Fixed read input method to be compatible with I/O technique script.

Cohii2 committed:
Changed taskList to static variable

Cohii2 committed:
Task Params Enum

Added an enumerator class to track parameters used in Task sub-classes.

Cohii2 committed:
Implemented MissingParamsException

Created new MissingParamsException Class, Enum array of TaskParams, methods to print missing parameters when exception is called.

Cohii2 committed:
Implemented InvalidCommandException

Cohii2 committed:
Fixed bug

Fixed bug where if user inputs "/command" without any argument, program throws an IndexOutOfBounds exception. Now, program will just read it in as blank input.

Cohii2 committed:
Updated ui-testing script

Updated testing script to test new exception printing.

Cohii2 committed:
Merge branch 'branch-Level-5'

Cohii2 committed:
Packages

Organised classes into suitable packages. Separated Main into new file.

Cohii2 committed:
Edited comment

Edited comment in Event class for clarity.

Cohii2 committed:
Created new deleteTask function

New deleteTask function for deleting tasks from list.

Cohii2 committed:
Added delete functionality.

Added ability to delete tasks from task list.

Cohii2 committed:
Updated IO Testing

Updated IO files and script to test delete function.

Cohii2 committed:
Save Tasks

Added functionality to save tasks in tasks.txt file.

Cohii2 committed:
Commented load functions

Added comments to load functions to clarify usage.

Cohii2 committed:
Merge branch 'branch-Level-6'

Cohii2 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/list.java

Cohii2 committed:
Create JAR file

Created JAR file.

NeoMinWei commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,14 @@
+import duke.*;
```
please import explicitly instead of using *. Check your other imports as well with the same issue

NeoMinWei commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+        super(description);
+        this.by = by;
+
+        if(hasMissingParams()) {
```
```suggestion
        if (hasMissingParams()) {
```

NeoMinWei commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+                this.getStatusIcon(), this.description, this.by);
+    }
+
+    /**
+     * Checks for parameters object is missing.
+     * Returns an array of missing parameters.
+     *
+     * @return List of missing parameters.
+     */
```
Good job on doing JavaDoc

NeoMinWei commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,36 @@
+        }
+    }
+
+    public static class EndListException extends Exception {
+
+    }
```
Why is this empty

NeoMinWei commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,60 @@
+        this.start = start;
+        this.end = end;
+
+        if(hasMissingParams()) {
```
```suggestion
        if (hasMissingParams()) {
```
please check other areas of your code for missing spaces

NeoMinWei commented on src/main/java/duke/command.java:
```diff
@@ -0,0 +1,168 @@
+        printLine();
+        for(int i = 0; i < taskList.size(); i++){
+            Task task = taskList.get(i);
+            System.out.println((i + 1) + "." + task);
```
why +1? please do not use magic numbers

NeoMinWei commented on src/main/java/duke/load.java:
```diff
@@ -0,0 +1,131 @@
+                break;
+            }
+        } catch (MissingParamsException ignored) {
+
```
try not to leave empty catch blocks

NeoMinWei commented on src/main/java/duke/load.java:
```diff
@@ -0,0 +1,131 @@
+                entry = String.format("%s|%s|%s" + System.lineSeparator(),
+                        "todo", task.getDescription(),
+                        task.getStatusIcon());
+            }
+            else if (task instanceof Deadline) {
```
else if is supposed to be on previous line, check your other else-ifs as well

NeoMinWei commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,230 @@
 Hello from
```
good job on doing text-ui-test

NeoMinWei reviewed (commented):
Good job on doing JavaDoc. There are certain coding standard violations and improvements to code quality that you need to take note of. Otherwise your code looks good

Cohii2 committed:
Fix IO test bug

Fixed IO test bug where IO test cannot create tasks.txt database

Cohii2 committed:
Create Storage Class

Cohii2 committed:
Create Ui Class

Cohii2 committed:
Created TaskList Class

Cohii2 committed:
Change comment

Cohii2 committed:
Create Parser Class

Cohii2 committed:
Create find function

Give users a way to find a task by searching for a keyword.

Cohii2 committed:
Added find function to Parser Class

Cohii2 committed:
Update IO testing for find function

Cohii2 committed:
Merge pull request #1 from Cohii2/branch-Level-9

Add find function

Cohii2 committed:
Add JavaDoc comments to the code.

Cohii2 committed:
Merge pull request #2 from Cohii2/branch-A-JavaDoc

Add JavaDoc comments

Cohii2 committed:
Add a User Guide

Add a User Guide to the project
